src/patch/apply.js

```diff
@@ -42,7 +40,61 @@ export function applyPatch(source, uniDiff, options = {}) {
             return false;
           }
         }
+        toPos++;
       }
+    }
+
+    return true;
+  }
+
+ // Search best fit offsets for each hunk based on the previous ones
+ for (let i = 0; i < hunks.length; i++) {
+    let hunk = hunks[i],
+        outOfLimits = 0,
+        localOffset = 0,
+        minLine = 0,
+        toPos = offset + hunk.oldStart - 1;
+
+    for (;;) {
```

Comment:
It's an optimization, the alternative is to have the old code and use a conditional to check it, or do the check twice on the most common case (the hunk fits at the first try, when `localOffset` is zero). This has only moved the block to check the current or next line before the loop, so it's checked always in a loop rollback fashion. The same would be achieved by doing the check previously to enter the loop, but code is duplicated. I can add more comments if you don't see it clear.